json_annotation/lib/src/json_value.dart

```diff
@@ -9,5 +9,10 @@ class JsonValue {
   /// Can be a [String] or an [int].
   final dynamic value;
 
-  const JsonValue(this.value);
+  /// Optional values that can be used when deserializing.
+  ///
+  /// The elements of [aliases] must be either [String] or [int].
+  final List<Object?> aliases;
```

Comment:
This should probably be a `Set` – since we don't care about order and we shouldn't allow duplicates.

Reply:
Done.

Reply:
Done.